docs/content/integrations/embedded-elt.mdx

```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
-    group_name="etl",
-    description="ETL Test",
-    deps=["foo"],
+```python file=/integrations/embedded_elt/sling_connection_resources.py
+from dagster_embedded_elt.sling.resources import (
+    SlingConnectionResource,
+    SlingResource,
 )
 
-asset_def = build_sling_asset(
-    asset_spec=asset_spec,
-    source_stream="file://path/to/file.csv",
-    target_object="main.dest_tbl",
-    mode=SlingMode.INCREMENTAL,
-    primary_key="id",
+from dagster import EnvVar
+
+sling_resource = SlingResource(
+    connections=[
+        # Using a connection string from an environment variable
+        SlingConnectionResource(
+            name="MY_POSTGRES",
+            type="postgres",
+            connection_string=EnvVar("POSTGRES_CONNECTION_STRING"),
+        ),
+        # Using a hard-coded connection string
+        SlingConnectionResource(
+            name="MY_DUCKDB",
+            type="duckdb",
+            connection_string="duckdb:///var/tmp/duckdb.db",
+        ),
+        # Using a keyword-argument constructor
+        SlingConnectionResource(
+            name="MY_SNOWFLAKE",
+            type="snowflake",
+            host=EnvVar("SNOWFLAKE_HOST"),
+            user=EnvVar("SNOWFLAKE_USER"),
+            role="REPORTING",
+        ),
+    ]
 )
+```
+
+## Step 3: Define the Sling assets
+
+Now you can define a Sling asset using the <PyObject module="dagster_embedded_elt.sling" object="sling_assets" /> decorator. Dagster will read the replication configuration to produce Assets.
+
+Each stream will render two assets, one for the source stream and one for the target destination. You may override how assets are named by passing in a custom <PyObject module="dagster_embedded_elt.sling" object="DagsterSlingTranslator" /> object.
 
-sling_job = build_assets_job(
-    "sling_job",
-    [asset_def],
-    resource_defs={"sling": sling_resource},
+```python file=/integrations/embedded_elt/sling_dagster_translator.py
+from dagster_embedded_elt import sling
+from dagster_embedded_elt.sling import (
+    DagsterSlingTranslator,
+    SlingResource,
+    sling_assets,
 )
 
+from dagster import Definitions, file_relative_path
+
+replication_config = file_relative_path(__file__, "../sling_replication.yaml")
+sling_resource = SlingResource(connections=[...])  # Add connections here
+
+
+@sling_assets(replication_config=replication_config)
+def my_assets(context, sling: SlingResource):
+    yield from sling.replicate(
+        replication_config=replication_config,
+        dagster_sling_translator=DagsterSlingTranslator(),
+    )
+    for row in sling.stream_raw_logs():
+        context.log.info(row)
+
+
+defs = Definitions(
+    assets=[
+        my_assets,
+    ],
+    resources={
+        "sling": sling_resource,
+    },
+)
 ```
 
----
+That's it! You should now be able to view your assets in Dagster and run the replication job.
 
 ## Examples
 
 This is an example of how to setup a Sling sync between Postgres and Snowflake:
```

Comment:
Thoughts on adding subheadings to these examples to make it easier to tell what they are?

Reply:
ooh i like it! picasso